x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RequestXContent.java

```diff
@@ -46,10 +46,12 @@ final class RequestXContent {
         PARAM_PARSER.declareString(constructorArg(), TYPE);
     }
 
-    private static final ParseField QUERY_FIELD = new ParseField("query");
+    // TODO: Maybe esql_version would be better? This looks like there's a JSON attribute `esql` with subattribute `version`.
```

Comment:
yeah, we should probably think about that a bit.  Worth discussing with Kibana as well.  For future proofing, I wonder if we want to set up a whole esql object, something like 

```
"esql": {
    "version": "2024.04.🎉 ",
    "whatever_future_setting": "goes_here"
}
```

Reply:
Discussed with the team: probably just `{"query": "...", "version": "2024.04.01.🎉"}` is fine.